src/GameLogic/PlayerActions/Craftings/DinorantCrafting.cs

```diff
@@ -0,0 +1,87 @@
+﻿// <copyright file="DinorantCrafting.cs" company="MUnique">
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.
+// </copyright>
+
+namespace MUnique.OpenMU.GameLogic.PlayerActions.Craftings;
+
+using System.Collections;
+using MUnique.OpenMU.DataModel.Configuration.ItemCrafting;
+using MUnique.OpenMU.DataModel.Configuration.Items;
+using MUnique.OpenMU.GameLogic.Attributes;
+using MUnique.OpenMU.GameLogic.PlayerActions.Items;
+using MUnique.OpenMU.GameLogic.Views.NPC;
+
+/// <summary>
+/// Crafting for Dinorant.
+/// </summary>
+public class DinorantCrafting : SimpleItemCraftingHandler
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DinorantCrafting"/> class.
+    /// </summary>
+    /// <param name="settings">The settings.</param>
+    public DinorantCrafting(SimpleCraftingSettings settings)
+        : base(settings)
+    {
+    }
+
+    /// <inheritdoc />
+    protected override CraftingResult? TryGetRequiredItems(Player player, out IList<CraftingRequiredItemLink> items, out byte successRate)
+    {
+        var craftingResult = base.TryGetRequiredItems(player, out items, out successRate);
+
+        var uniriaLink = items.Where(i => i.ItemRequirement.PossibleItems.Any(i => i.Name == "Horn of Uniria"));
+        foreach (var item in uniriaLink.First().Items)
+        {
+            if (item.Durability < 255)
+            {
+                return CraftingResult.IncorrectMixItems;
+            }
+        }
+
+        return craftingResult;
+    }
+
+    /// <inheritdoc/>
+    protected override void AddRandomItemOption(Item resultItem, Player player, byte successRate)
+    {
+        if (Rand.NextRandomBool(30)
+            && resultItem.Definition!.PossibleItemOptions.FirstOrDefault(o =>
+                    o.PossibleOptions.Any(p => p.OptionType == ItemOptionTypes.Option))
+                is { } option)
+        {
+            var link = player.PersistenceContext.CreateNew<ItemOptionLink>();
+            link.ItemOption = option.PossibleOptions.SelectRandom();
+            resultItem.ItemOptions.Add(link);
+
+            // There is a second rollout for an additional bonus option to the first (but only if it doesn't coincide).
+            if (Rand.NextRandomBool(20))
+            {
+                var bonusOpt = option.PossibleOptions.SelectRandom();
+                if (bonusOpt != link.ItemOption)
+                {
+                    var bonusLink = player.PersistenceContext.CreateNew<ItemOptionLink>();
+                    bonusLink.ItemOption = bonusOpt;
+                    resultItem.ItemOptions.Add(bonusLink);
+                }
+            }
+        }
```

Comment:
A 2-option dino was quite the rare thing, but I do remember them existing! Also found an old MUHQ thread selling one.
![Screenshot from 2024-12-09 11-09-49](https://github.com/user-attachments/assets/6f725970-4a92-4406-8512-8eeaa6128746)